### ISSUE TITLE
Cannot read property `.get` of undefined

### ISSUE THREAD

**Issue by iway1** [supertokens/supertokens-react-native#124]
Sometimes we're seeing an error in our logging that reads `Cannot read property 'get' of undefined`, which is happening in this SDK code. 

I believe there is a race condition such that the axios client is sometimes defined when calling `.get` somewhere in the supertokens react native SDK because it happens randomly. Not sure how this could happen, or whether it represents the library breaking entirely, but it is worrying. Please advise

**Comment by rishabhpoddar** [comment]
Hi @iway1 

Could you please share the error stack along with which version of our SDK you are using? 

Also, have you tried to upgrade to the latest version our of lib and seen if the error comes in there?

**Comment by iway1** [comment]
@rishabhpoddar stack trace:

```
SharedConstantsProviderTermsAndConditionsTypeError: Cannot read property 'get' of undefined
  at ?anon_0_(/home/expo/workingdir/build/node_modules/supertokens-react-native/lib/build/axiosError.js:77:49)
  at next(native)
  at anonymous(node_modules/tslib/tslib.js:169:79)
  at tryCallTwo(node_modules/promise/setimmediate/core.js:45:7)
  at doResolve(node_modules/promise/setimmediate/core.js:200:23)
  at Promise(node_modules/promise/setimmediate/core.js:66:12)
  at __awaiter(node_modules/tslib/tslib.js:165:39)
  at createAxiosErrorFromFetchResp(/home/expo/workingdir/build/node_modules/supertokens-react-native/lib/build/axiosError.js:72:21)
  at ?anon_0_(/home/expo/workingdir/build/node_modules/supertokens-react-native/lib/build/axios.js:319:78)
  at next(native)
  at fulfilled(node_modules/tslib/tslib.js:166:66)
  at tryCallOne(node_modules/promise/setimmediate/core.js:37:14)
  at anonymous(node_modules/promise/setimmediate/core.js:123:25)
  at apply(native)
  at anonymous(node_modules/react-native/Libraries/Core/Timers/JSTimers.js:247:23)
  at _callTimer(node_modules/react-native/Libraries/Core/Timers/JSTimers.js:111:15)
  at _callReactNativeMicrotasksPass(node_modules/react-native/Libraries/Core/Timers/JSTimers.js:161:15)
  at callReactNativeMicrotasks(node_modules/react-native/Libraries/Core/Timers/JSTimers.js:415:42)
  at __callReactNativeMicrotasks(node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:393:44)
  at anonymous(node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:132:39)
  at __guard(node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:368:11)
  at flushedQueue(node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:131:17)
  at invokeCallbackAndReturnFlushedQueue(node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:127:29)
```

Source map says it's this:

![Screenshot 2024-06-06 at 8 58 23 AM](https://github.com/supertokens/supertokens-react-native/assets/12774588/120df9b2-783c-43ed-809b-b0afac02c8d4)

**Comment by iway1** [comment]
> Also, have you tried to upgrade to the latest version our of lib and seen if the error comes in there?

No I haven't tried it, is there any reason to believe this would be fixed in the latest version? There's nothing in the change logs about it

**Comment by rishabhpoddar** [comment]
Thanks for the stack. You need not update, however, which version are you using? So we can make sense of the stack trace.

**Comment by iway1** [comment]
@rishabhpoddar  `4.0.4`. Thanks!

**Comment by rishabhpoddar** [comment]
Thanks! We will check this out in the coming and fix it!

**Comment by rishabhpoddar** [comment]
@iway1 this has been fixed in version 5.1.1 of our react native SDK. Keeping this issue open because we have to also fix this in supertokens-website repo.
